src/java.compiler/share/classes/javax/lang/model/element/ElementKind.java

```diff
@@ -121,7 +123,14 @@ public enum ElementKind {
      * A binding variable in a pattern.
      * @since 16
      */
-    BINDING_VARIABLE;
+    BINDING_VARIABLE,
+
+    /**
+     * A local component variable in a derived record creation expression.
+     * @since 23
+     */
+    @PreviewFeature(feature=PreviewFeature.Feature.DERIVED_RECORD_CREATION, reflective=true)
+    COMPONENT_LOCAL_VARIABLE;
```

Comment:
We could, and I was thinking of that. But, then I decided to go with a new kind, as I think it matches better the overall design of the Elements/Kinds. E.g. we have existing `EXCEPTION_PARAMETER` (used in `catch`) or `BINDING_VARIABLE`, rather than using `LOCAL_VARIABLE` for them.

(Also, many of these variable kinds have some small, but significant differences in semantics from `LOCAL_VARIABLE` - e.g. some are never blank/are always definitely assigned.)